### ISSUE TITLE
tsql function keyword after select makes a cycle of LT09 LT14 LT02 under specific conditions

### ISSUE THREAD

**Issue by mmoole** [sqlfluff/sqlfluff#8389]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

in
```sql
...
  -- cross apply for finding the tags for tags
  CROSS APPLY (
    SELECT STRING_AGG(T1.NAME, ',') AS TAGS_LIST
    FROM
      OPENJSON (C.TAGS, '$')
...
```
the function `STRING_AGG` should begin in a new line. Sqlfluff does this and has rules for this, but if this is the case, another rule takes over for the next step, and after this yet another, which result in the first rule being applied, thus the cycle is complete. See animated gif for an example below in 'observed behaviour'.

### Expected Behaviour

New line and indent should have been applied :

``` sql 
  CROSS APPLY (
    SELECT 
      STRING_AGG(T1.NAME, ',') AS TAGS_LIST
    FROM
      OPENJSON (C.TAGS, '$')
      WITH

```

### Observed Behaviour

There is a cycle of the rules LT09 LT14 LT02:

<img width="960" height="640" alt="Image" src="https://github.com/user-attachments/assets/bca8150f-13d6-492d-b25e-a9e44bd0ee82" />

### How to reproduce

here is the full query:

``` sql
SET

  @Markdown = @Html;

SELECT
  C.CUSTOMER_ID,
  COALESCE(SHORT_TAGS.TAGS_LIST, '') AS 'Contexts',
  CASE
    WHEN O.ORDER_COUNT > 10 THEN 'large' ELSE 'small'
  END AS CUSTOMER_TYPE
FROM
  CUSTOMERS AS C
  LEFT JOIN ORDERS AS O
    ON C.CUSTOMER_ID = O.CUSTOMER_ID

  -- cross apply for finding the tags for tags
  CROSS APPLY (
    SELECT 
      STRING_AGG(T1.NAME, ',') AS TAGS_LIST
    FROM
      OPENJSON (C.TAGS, '$')
      WITH
      (GID_TAG BIGINT '$.gid') AS TAG_TABLE
      INNER JOIN THE_TAGS AS T1
        ON TAG_TABLE.GID_TAG = T1.GID
    WHERE
      T1.NAME LIKE '#%'
  ) AS SHORT_TAGS

ORDER BY O.ORDER_COUNT DESC, C.CUSTOMER_ID ASC;

```

### Dialect

tsql

### Version

sqlfluff --version                                                                                                                                                                                                   
sqlfluff, version 4.3.0

### Configuration

.sqlfluff is empty

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by devYRPauli** [comment]
I cannot reproduce the cycle from the CLI. I ran your full query on released 4.3.0 and on main at d3e8358, both with an empty `.sqlfluff`.

`sqlfluff fix --dialect tsql` changes the file on pass 1 and changes nothing on pass 2. `sqlfluff lint --dialect tsql` on the result reports `All Finished!` with zero violations. I also fixed with `--rules LT09`, `--rules LT14` and `--rules LT02` one at a time, in that order, for four rounds. LT09 fires once, LT02 fires once, and rounds 2 to 4 change nothing.

What I do see is the behaviour behind your screenshot. Your query lints as:

```
L:  18 | P:   5 | LT09 | Select targets should be on a new line unless there is
                       | only one select target. [layout.select_targets]
```

The `SELECT` inside `CROSS APPLY` has exactly one select target, `STRING_AGG(T1.NAME, ',') AS TAGS_LIST`. The default `single_target_policy = same_line` asks for a single target on the same line as `SELECT`. So LT09 pulls `STRING_AGG` up, which is the opposite of the layout you want.

Set the policy to get your expected output:

```ini
[sqlfluff:rules:layout.select_targets]
single_target_policy = new_line
```

With that config, `fix` gives you this and stabilises on pass 2:

```sql
CROSS APPLY (
    SELECT
        STRING_AGG(T1.NAME, ',') AS TAGS_LIST
    FROM
        OPENJSON (C.TAGS, '$')
        WITH
        (GID_TAG BIGINT '$.gid') AS TAG_TABLE
    INNER JOIN THE_TAGS AS T1
        ON TAG_TABLE.GID_TAG = T1.GID
    WHERE
        T1.NAME LIKE '#%'
) AS SHORT_TAGS
```

I measured that on released 4.3.0 as well, so you do not need a new version.

What runs `fix` in the recording? A fix-on-save loop in an editor extension applies rules on each save, and it can look like a cycle when each save re-edits the file. If you get the cycle from a plain `sqlfluff fix` in a shell, post the exact command and the `.sqlfluff` file. I will look again.

**Comment by mmoole** [comment]
hello, the observed behavior gif is indeed vscode and the sqlfluff extension using a zero .sqlfluff config for formatting via alt+shift+f, and having it set to tsql as flavour.
But interestingly, I also cannot reproduce this any more!
I suspect something was finally really mixed up, as I am redoing the test with the same version 4.3.0 of sqlfluff, and it works the same as in your test. I also threw in some or all of my config and commented out block by block, but was not able to reproduce the cyclic behaviour.
So to sum it up, the issue can be closed as a non-issue I think 😇

**Comment by devYRPauli** [comment]
Agreed, this is not a cycle.

For anyone who finds this later: the behaviour in the recording is LT09 with the default `single_target_policy = same_line`. That policy pulls a lone select target up onto the `SELECT` line. Setting `single_target_policy = new_line` under `[sqlfluff:rules:layout.select_targets]` gives the layout you expected. I measured that on 4.3.0 and on main at d3e8358.
